components/wmtk_components/marching/internal/Marching.cpp

```diff
@@ -0,0 +1,112 @@
+#include "Marching.hpp"
+
+#include <wmtk/SimplicialComplex.hpp>
+#include <wmtk/operations/tri_mesh/EdgeCollapseToMidpoint.hpp>
+#include <wmtk/operations/tri_mesh/EdgeSplitAtMidpoint.hpp>
+#include <wmtk/operations/tri_mesh/EdgeSplitWithTag.hpp>
+#include <wmtk/operations/tri_mesh/EdgeSwap.hpp>
+#include <wmtk/operations/tri_mesh/FaceSplitWithTag.hpp>
+#include <wmtk/operations/tri_mesh/VertexTangentialLaplacianSmooth.hpp>
+
+namespace wmtk::components::internal {
+
+Marching::Marching(
+    TriMesh& mesh,
+    MeshAttributeHandle<double>& position_handle,
+    MeshAttributeHandle<long>& vertex_tag,
+    MeshAttributeHandle<long>& edge_tag,
+    const long input_tag_value,
+    const long embedding_tag_value,
+    const long split_tag_value,
+    const int dimension,
+    const bool lock_boundary)
+    : m_mesh(mesh)
+    , m_position_handle(position_handle)
+    , m_vertex_tag(vertex_tag)
+    , m_edge_tag(edge_tag)
+    , m_input_tag_value(input_tag_value)
+    , m_embedding_tag_value(embedding_tag_value)
+    , m_split_tag_value(split_tag_value)
+    , m_dimension(dimension)
+    , m_scheduler(mesh)
+    , m_lock_boundary(lock_boundary)
+{}
+
+void Marching::process()
+{
+    switch (m_dimension) {
+    case 2: process_in_2d(); break;
+    case 3: process_in_3d(); break;
+    default: throw std::runtime_error("settings went wrong!"); break;
+    }
+}
+
+void Marching::process_in_2d()
+{
+    using namespace operations;
+
+    wmtk::MeshAttributeHandle<long> todo_edgesplit_same_handle = m_mesh.register_attribute<long>(
+        std::string("todo_edgesplit_same_tag"),
```

Comment:
You don't need to write `std::string("...")` you can just write `"..."` directly.

I guess the name should be `"todo_edgesplit_same_handle"`?